woocommerce/wc-admin
# TRY fixing persisted queries by moving NavigationComponent back out of the router HoC

Fixes #6581 

As per the original issue discussion, this PR employs one possible fix for the fact that persisted queries are no longer working as expected in the new navigation feature. 

In short, there was a change made in an [unrelated PR](https://github.com/woocommerce/woocommerce-admin/pull/6247) that resulted in the `NavigationPlugin` component being remounted on every router change, which causes the component to lose state that was being used to implement the persisted queries.

This PR is one potential solution for that issue, which simply involves moving the mentioned component _back_ to where it came from, which is it's own React root on a custom DOM element. 

I don't _believe_ this diminishes the utility of the PR that moved it, but I'm paging @joshuatf to ensure that that's the case. If so, we could possibly try a solution that would involve persisting state via localStorage/etc as mentioned in the originating issue.

### Detailed test instructions:

1. Go to Analytics Report
2. Change Time period
3. Navigate to homescreen
4. Navigate back to Analytics Report
5. Notice that the time period is _still_ what you set on step 2
